jOOQ/src/main/java/org/jooq/DSLContext.java

```diff
@@ -9660,15 +9660,15 @@
      *
      * @see DSL#dropTemporaryTable(String)
      */
-    @Support
+    @Support({ MYSQL })
     DropTableStep dropTemporaryTable(String table);
 
     /**
      * Create a new DSL <code>DROP TEMPORARY TABLE</code> statement.
      *
      * @see DSL#dropTemporaryTable(Name)
      */
-    @Support
+    @Support({ MYSQL })
```

Comment:
I'm not sure about this change. This would mean that *only* `MYSQL` supports the syntax, which would be backwards incompatible, and does not correspond to your suggested implementation. I'd prefer to keep the annotation as it is for now and fix this separately, as we validate other dialects in the future.